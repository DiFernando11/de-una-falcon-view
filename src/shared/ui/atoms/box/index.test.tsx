import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Box from '.';

describe('<Box />', () => {
  it('should match snapshot', () => {
    // Arrange
    const content = <span>Hola mundo</span>;
    // Act
    const { asFragment } = render(<Box>{content}</Box>);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    // Arrange
    const content = <span>Hola mundo</span>;
    // Act
    render(<Box>{content}</Box>);
    // Assert
    expect(screen.getByText('Hola mundo')).toBeInTheDocument();
  });

  it('should apply custom className', () => {
    // Arrange
    const className = 'test-class';
    const content = 'Contenido';
    // Act
    render(<Box className={className}>{content}</Box>);
    const div = screen.getByText(content);
    // Assert
    expect(div).toHaveClass(className);
  });

  it('should pass other props to the div', () => {
    // Arrange
    const testId = 'box-test';
    const ariaLabel = 'box-label';
    const content = 'Prueba';
    // Act
    render(
      <Box data-testid={testId} aria-label={ariaLabel}>
        {content}
      </Box>
    );
    const box = screen.getByTestId(testId);
    // Assert
    expect(box).toHaveAttribute('aria-label', ariaLabel);
  });
});
