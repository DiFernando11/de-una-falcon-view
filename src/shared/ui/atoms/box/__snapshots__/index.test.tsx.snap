// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Box /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      Hola mundo
    </span>
  </div>
</DocumentFragment>
`;
